src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -1115,7 +1121,55 @@ private void GenerateLayoutGraph(GraphLayout.Graph graph)
             }
         }
 
-        private void RunLayoutGraph(GraphLayout.Graph graph)
+        public void GenerateSeparateSubgraphs()
+        {
+            int processed = 0;
+            var combinedGraph = Model.LayoutSubgraphs.First();
+            GraphLayout.Graph graph = graph = new GraphLayout.Graph();
+            Queue<GraphLayout.Node> queue = new Queue<GraphLayout.Node>();
+
+            while (combinedGraph.Nodes.Count(n => n.IsSelected) > 0)
+            {
+                GraphLayout.Node currentNode;
+
+                if (queue.Count == 0)
+                {
+                    if (graph.Nodes.Count > 0)
+                    {
+                        graph.ParentGraph = combinedGraph;
+                        Model.LayoutSubgraphs.Add(graph);
+                        combinedGraph.Nodes.ExceptWith(graph.Nodes);
+                        graph = new GraphLayout.Graph();
+                    }
+                    if (combinedGraph.Nodes.Count(n => n.IsSelected) == 0) break;
+
+                    currentNode = combinedGraph.Nodes.FirstOrDefault(n => n.IsSelected);
+                    graph.Nodes.Add(currentNode);
+                }
+                else
+                {
+                    currentNode = queue.Dequeue();
+                }
+
+                var selectedNodes = currentNode.RightEdges.Select(e => e.EndNode)
```

Comment:
Maybe you want to add some comments here on what is it that you're doing here. Dense Linq expressions can be quite hard to decipher at times.